util/types/convert.go

```diff
@@ -97,19 +97,23 @@ func convertIntToInt(val, lowerBound, upperBound int64, tp byte) (converted int6
 	return
 }
 
+func convertUintToInt(val uint64, lowerBound, upperBound int64, tp byte) (converted int64, err error) {
+	if val > uint64(upperBound) {
+		converted = upperBound
+		err = overflow(val, tp)
+		return
```

Comment:
Yes, that seems clearer.
I only remove code from convertToInt function, there are some places have this problem, so i prefer to do it in next PR.